inc/Engine/Optimization/LazyRenderContent/Frontend/Controller.php

```diff
@@ -42,14 +42,15 @@ public function __construct( Processor $processor, ContextInterface $context ) {
 	 * @return string
 	 */
 	public function optimize( string $html, $row ): string {
+		$html_without_hashes = $this->remove_hashes( $html );
```

Comment:
@Khadreal Is it possible to run this line only when needed? Here I think we risk running it even in cases where `$html_without_hashes` won't be used and since it involves a regex (that can be huge in some cases), it would be safer for performances to avoid unnecessary computations.

Reply:
Yes, we could but every return in this method need to remove the hashed before returning.
We could move it inside the logic, which means `remove_hashes` will be repeated multiple times.

so we we'll have something like this 
```
if ( ! $row->has_lrc() ) {
			return $this->remove_hashes( $html );
		}
		$hashes = json_decode( $row->below_the_fold );

		if ( null === $hashes || ! is_array( $hashes ) ) {
			return $this->remove_hashes( $html );
		}

		$result = preg_replace( '/data-rocket-location-hash="(?:' . implode( '|', $hashes ) . ')"/i', 'data-wpr-lazyrender="1"', $html, -1, $count );

		if (
			null === $result
			||
			0 === $count
		) {
			return $this->remove_hashes( $html );
		}
		........
```

Reply:
I think that's OK to repeat the method 👍